src/quest_manager/models.py

```diff
@@ -19,12 +19,52 @@
 from tags.models import TagsModelMixin
 
 
+class CategoryQuerySet(models.QuerySet):
+    """
+    Custom queryset for Category model (campaigns).
+    """
+    def with_available_quests(self):
+        """
+        Annotates and filters categories that are active and have at least one
+        quest that is visible to students and not archived.
+        Uses a correlated subquery with EXISTS for performance.
+        """
+        filtered_quests = Quest.objects.filter(
+            campaign=OuterRef('pk'),
+            visible_to_students=True,
+            archived=False
+        )
+
+        return self.filter(active=True).annotate(
+            has_available_quests=Exists(filtered_quests)
+        ).filter(has_available_quests=True)
+
+
+class CategoryManager(models.Manager):
+    """
+    Custom model manager for Category (campaign) objects.
+    """
+    def get_queryset(self):
+        """
+        Returns the base QuerySet for Category objects,
+        using the custom CategoryQuerySet with additional filtering utilities.
+        """
+        return CategoryQuerySet(self.model, using=self._db)
+
+    def all_active_with_available_quests(self):
```

Comment:
Just put all the code here. I don't think we need  CategoryQuerySet at this point, so simplify the code and just put it all in a single Manager method.